docker/rosetta.Dockerfile

```diff
@@ -0,0 +1,300 @@
+FROM debian:buster
+ARG STACKS_NETWORK
+ARG PG_VERSION
+ARG PG_VERSION
+ARG PG_HOST
+ARG PG_PORT
+ARG PG_USER
+ARG PG_PASSWORD
+ARG SEED_CHAINSTATE
+ARG STACKS_API_VERSION
+ARG STACKS_BLOCKCHAIN_VERSION
+ENV SEED_CHAINSTATE=${SEED_CHAINSTATE}
+ENV STACKS_API_VERSION=${STACKS_API_VERSION}
+ENV STACKS_BLOCKCHAIN_VERSION=${STACKS_BLOCKCHAIN_VERSION}
+ENV PG_VERSION=${PG_VERSION}
+ENV PG_HOST=${PG_HOST}
+ENV PG_PORT=${PG_PORT}
+ENV PG_USER=${PG_USER}
+ENV PG_PASSWORD=${PG_PASSWORD}
+ENV PG_DATABASE=stacks_blockchain_api
+ENV PGDATA=/postgres
+ENV PG_SCHEMA=stacks_blockchain_api
+ENV STACKS_BLOCKCHAIN_DIR=/stacks-blockchain
+ENV STACKS_BLOCKCHAIN_API_DIR=/stacks-blockchain-api
+ENV STACKS_NETWORK=${STACKS_NETWORK}
+ENV STACKS_CORE_EVENT_PORT=3700
+ENV STACKS_CORE_EVENT_HOST=127.0.0.1
+ENV STACKS_EVENT_OBSERVER=127.0.0.1:3700
+ENV STACKS_BLOCKCHAIN_API_PORT=3999
+ENV STACKS_BLOCKCHAIN_API_HOST=0.0.0.0
+ENV STACKS_CORE_RPC_HOST=127.0.0.1
+ENV STACKS_CORE_RPC_PORT=20443
+ENV STACKS_CORE_P2P_PORT=20444
+ENV LANG en_US.UTF-8  
+ENV LANGUAGE en_US:en  
+ENV LC_ALL en_US.UTF-8
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update \
+    && apt-get install -y \
+        gnupg2 \
+        lsb-release \
+        curl \
+        jq \
+        procps \
+        netcat \
+        gosu \
+        locales
+RUN sed -i '/en_US.UTF-8/s/^# //g' /etc/locale.gen && locale-gen
+
+RUN curl -sL https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add - \
+    && echo "deb http://apt.postgresql.org/pub/repos/apt/ `lsb_release -cs`-pgdg main" > /etc/apt/sources.list.d/pgsql.list \
+    && curl -sL https://deb.nodesource.com/setup_16.x | bash -
+RUN apt-get update \
+    && apt-get install -y \
+        postgresql-${PG_VERSION} \
+        postgresql-client-${PG_VERSION} \
+        nodejs \
+    && echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
+RUN mkdir -p \
+        ${STACKS_BLOCKCHAIN_DIR}/data \
+        /scripts \
+    && apt-get clean \
+    && rm -rf /var/cache/apt/* /var/lib/apt/lists/* /tmp/*
+COPY --from=stacks-blockchain-build /out ${STACKS_BLOCKCHAIN_DIR}
+COPY --from=stacks-blockchain-api-build /app ${STACKS_BLOCKCHAIN_API_DIR}
+COPY --from=stacks-blockchain-build /src/testnet/stacks-node/conf/*follower-conf.toml ${STACKS_BLOCKCHAIN_DIR}/
+
+###################################
+## entrypoint.sh
+RUN <<EOF
+cat > /entrypoint.sh <<'EOM'
+#!/bin/bash -e
+exec 2>&1
+# enable json logging for stacks-blockchain
+export STACKS_LOG_JSON=1
+# configure postgres and start it
+mkdir -p "${PGDATA}" || exit 1
+chown -R postgres:postgres "${PGDATA}" || exit 1
+gosu postgres /usr/lib/postgresql/${PG_VERSION}/bin/initdb -D "${PGDATA}" --wal-segsize=512  || exit 1
+echo "host all all all trust" >> "$PGDATA/pg_hba.conf" || exit 1
+gosu postgres /usr/lib/postgresql/${PG_VERSION}/bin/pg_ctl start -W -D ${PGDATA} -o "-c listen_addresses='*'"
+# give postgres 10s to startup
+echo "waiting 10s for postgres to start"
+sleep 10
```

Comment:
I think if you change the `pg_ctl start` flag `-W` to `-w` (lowercase) then it will wait correctly for startup, and you could remove this sleep.

Reply:
Indeed it does!  I had changed that line from a `postmaster` exec (where -w doesn't work), but i like this idea - i'll have to re-test it, but shouldn't be an issue to update. for whatever reason, `-w` when using postmaster was throwing an error.

Reply:
updated and tested with testnet, all good